### PR TITLE
refactor(manager): Move contructor's parameter wallet_index and utxo_index to builder

### PR DESCRIPTION
**Pull request by msbrogli**
## Acceptance Criteria

1) Remove `wallet_index` and `utxo_index` from `HathorManager`'s constructor.
2) Refactor tests to use `self.create_peer(...)` instead of instantiating `HathorManager` directly.
3) Remove `TransactionStorage.with_index` attribute as a preparation to move the `IndexManager` instantiation to the builder.

### CODE REVIEW

File: tests/others/test_metrics.py [hunk 1]
```diff
@@ -60,11 +58,9 @@ def test_connections_manager_integration(self):
         self.tmpdirs.append(tmpdir)
         wallet = Wallet(directory=tmpdir)
         wallet.unlock(b'teste')
-        manager = HathorManager(self.clock, tx_storage=tx_storage, wallet=wallet)
+        manager = self.create_peer('testnet', tx_storage=tx_storage, wallet=wallet)
 
-        manager.metrics.start()
```
Comment (pedroferreira1) [inline tests/others/test_metrics.py:65]: Why you don't need this anymore?

**Review by pedroferreira1 — APPROVED** [review]

File: hathor/manager.py [hunk 2]
```diff
@@ -165,18 +158,9 @@ def __init__(self, reactor: Reactor, peer_id: Optional[PeerId] = None, network:
             self.checkpoints_ready[0] = True
 
         # XXX Should we use a singleton or a new PeerStorage? [msbrogli 2018-08-29]
-        self.pubsub = pubsub or PubSubManager(self.reactor)
```
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/manager.py:168]: Nice catch. Thanks! I deleted it.

File: hathor/builder.py [hunk 3]
```diff
@@ -131,16 +133,27 @@ def create_manager(self, reactor: PosixReactorBase, args: Namespace) -> HathorMa
...
 
+        pubsub = PubSubManager(reactor)
+
+        if args.wallet_index and tx_storage.indexes is not None:
+            assert tx_storage.indexes is not None
+            self.log.debug('enable wallet indexes')
+            self.enable_wallet_index(tx_storage.indexes, pubsub)
+
+        if args.utxo_index and tx_storage.indexes is not None:
+            assert self.tx_storage.indexes is not None
```
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder.py]: Fixed. Thanks!

File: hathor/builder.py [hunk 4]
```diff
@@ -131,16 +133,27 @@ def create_manager(self, reactor: PosixReactorBase, args: Namespace) -> HathorMa
         enable_sync_v1 = not args.x_sync_v2_only
         enable_sync_v2 = args.x_sync_v2_only or args.x_sync_bridge
 
+        pubsub = PubSubManager(reactor)
+
+        if args.wallet_index and tx_storage.indexes is not None:
+            assert tx_storage.indexes is not None
```
[the remark this answers is not in the document]
Comment (msbrogli) [reply hathor/builder.py]: It worked after removing the assert. Fixed. Thanks!

File: tests/others/test_metrics.py (hunk 1 above)
Comment (msbrogli) [reply tests/others/test_metrics.py:65]: We are using `self.create_peer(...)`, which runs `manager.start()`, which runs `manager.metrics.start()`.

**Pull request merged**